packages/alfa-media/src/media.ts

```diff
@@ -523,18 +1129,17 @@ export namespace Media {
    */
   const parseConditionWithoutOr = either(
     parseNot,
-    map(pair(parseInParens, zeroOrMore(parseAnd)), (result) => {
-      const [left, right] = result;
-      return [left, ...right].reduce((left, right) =>
+    map(pair(parseInParens, zeroOrMore(parseAnd)), ([left, right]) =>
+      [left, ...right].reduce((left, right) =>
         Condition.of(Combinator.And, left, right)
-      );
-    })
+      )
+    )
   );
 
   /**
    * @see https://drafts.csswg.org/mediaqueries/#media-query
    */
-  export class Query implements Equatable, Serializable {
+  export class Query implements Queryable {
     public static of(
       modifier: Option<Modifier>,
       type: Option<Type>,
```

Comment:
I'm really reluctant in keeping this as `type` because we do have a fairly consistent use of `type` as a discriminator field almost everywhere.
Especially, since unknown media types are accepted, it would mean that for example a media query `"length"`  would result in an object whose `type` is `length` but is absolutely not a `Length` 🙈 

This is somewhat tempered by typing itself (since stuff that are `Query | Length` are fairly rare), but this could happen with any other object and feels way too brittle for my confort zone.

Reply:
I don't agree. The term "type" has a very particular meaning for this structure and since we have no need to discriminate it as part of a union, "type" does seem like the best fit. The fact that the term "type" is always used as a discriminator when we deal with union types doesn't really seem relevant here as we're not dealing with a union. Additionally, the `Type` structure doesn't serialise to a `string` so there's never any doubt the the `Query#type` property _isn't_ a discriminator because you'd see this:

```js
{
  type: {
    name: "length"
  }
}
```